op-node/rollup/interop/managed/system_test.go

```diff
@@ -0,0 +1,252 @@
+		expectedResult uint64
+		expectedError  string
+	}{
+		{
+			name:           "target_equals_latest",
+			l2Unsafe:       100,
+			latestUnsafe:   100,
+			validBlocks:    []uint64{100},
+			expectedResult: 100,
+		},
+		{
+			name:           "all_blocks_valid",
+			l2Unsafe:       100,
+			latestUnsafe:   105,
+			validBlocks:    []uint64{100, 101, 102, 103, 104, 105},
+			expectedResult: 105,
+		},
+		{
+			name:           "all_blocks_invalid",
+			l2Unsafe:       100,
+			latestUnsafe:   105,
+			validBlocks:    []uint64{100}, // only l2Unsafe is valid
+			expectedResult: 100,
+		},
+		{
+			name:           "mixed_validity_case1",
+			l2Unsafe:       100,
+			latestUnsafe:   105,
+			validBlocks:    []uint64{100, 101, 102}, // 103-105 invalid
+			expectedResult: 102,
+		},
+		{
+			name:           "single_block_ahead_valid",
+			l2Unsafe:       100,
+			latestUnsafe:   101,
+			validBlocks:    []uint64{100, 101},
+			expectedResult: 101,
+		},
+		{
+			name:           "single_block_ahead_invalid",
+			l2Unsafe:       100,
+			latestUnsafe:   101,
+			validBlocks:    []uint64{100}, // 101 invalid
+			expectedResult: 100,
+		},
+		{
+			name:           "target_not_at_100",
+			l2Unsafe:       95,
+			latestUnsafe:   100,
+			validBlocks:    []uint64{95, 96, 97}, // 98-100 invalid
+			expectedResult: 97,
+		},
+		{
+			name:           "target_is_invalid",
+			l2Unsafe:       100,
+			latestUnsafe:   100,
+			validBlocks:    []uint64{96, 97, 98, 99}, // 96-99 valid
+			expectedResult: 99,
+		},
+		{
+			name:           "target_is_larger_than_latest",
+			l2Unsafe:       101,
+			latestUnsafe:   100,
+			validBlocks:    []uint64{96, 97, 98, 99}, // 96-99 valid
+			expectedResult: 99,
+		},
+		{
+			name:           "walkback_after_binary_search",
+			l2Unsafe:       95,
+			latestUnsafe:   105,
+			validBlocks:    []uint64{92, 93}, // 92-93 valid
+			expectedResult: 93,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			logger := testlog.Logger(t, log.LevelDebug)
+
+			mockL1 := &MockL1Source{}
+			mockL2 := &MockL2Source{}
+
+			// Setup l2Unsafe block
+			l2UnsafeHash := common.HexToHash(fmt.Sprintf("0x%x", tt.l2Unsafe))
+			l2UnsafeL1Origin := eth.BlockID{Hash: common.HexToHash("0xa"), Number: 10}
+			l2UnsafeBlockRef := createL2BlockRef(tt.l2Unsafe, l2UnsafeHash.Hex(), l2UnsafeL1Origin)
+
+			mockL2.On("L2BlockRefByHash", mock.Anything, l2UnsafeHash).Return(l2UnsafeBlockRef, nil)
+
+			// Setup latest unsafe block
+			latestHash := common.HexToHash(fmt.Sprintf("0x%x", tt.latestUnsafe))
+			latestL1Origin := eth.BlockID{Hash: common.HexToHash("0xf"), Number: 10 + tt.latestUnsafe - tt.l2Unsafe}
+			latestBlockRef := createL2BlockRef(tt.latestUnsafe, latestHash.Hex(), latestL1Origin)
+
+			mockL2.On("L2BlockRefByLabel", mock.Anything, mock.MatchedBy(func(label eth.BlockLabel) bool {
+				return label == eth.Unsafe
+			})).Return(latestBlockRef, nil)
```

Comment:
why doesn't
```suggestion
			mockL2.On("L2BlockRefByLabel", mock.Anything, eth.Unsafe).Return(latestBlockRef, nil)
```
work?

Reply:
I think because the `mocks` are a bit outdated, and were autogenerated before `eth.Unsafe` was introduced.